java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -174,6 +174,27 @@ protected boolean compareBaseVariableWidthVectors(BaseVariableWidthVector left)
       return false;
     }
 
+    final int offsetWidth = BaseVariableWidthVector.OFFSET_WIDTH;
+
+    // compare data buffers to make sure they have identical content
+    final int leftDataStart = left.getOffsetBuffer().getInt(leftStart * offsetWidth);
+    final int leftDataEnd = left.getOffsetBuffer().getInt((leftStart + length) * offsetWidth);
+    final int rightDataStart = right.getOffsetBuffer().getInt(rightStart * offsetWidth);
+    final int rightDataEnd = right.getOffsetBuffer().getInt((rightStart + length) * offsetWidth);
+
+    // check total length
+    if (rightDataEnd - rightDataStart != leftDataEnd - leftDataStart) {
+      return false;
+    }
+
+    int result = ByteFunctionHelpers.equal(left.getDataBuffer(), leftDataStart, leftDataEnd,
```

Comment:
are we assuming that there are no null values in between? after discussion with @pravindra i think we can have non-zero values in offset even if the index is null.

Reply:
If the data comes from a mmap buffer from C++/C, it can be uninitialized memory and no assumption should be made about the content.